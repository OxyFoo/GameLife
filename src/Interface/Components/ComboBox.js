import * as React from 'react';
import { Animated, FlatList, StyleSheet, TouchableHighlight, View } from 'react-native';

import Text from './Text';
import Button from './Button';
import Icon from './Icon';
import Input from './Input';
import themeManager from '../../Managers/ThemeManager';
import langManager from '../../Managers/LangManager';
import { SpringAnimation } from '../../Functions/Animations';

const ComboBoxProps = {
    style: {},
    activeColor: 'main1',
    title: 'Title',
    /**
     * @type {Array<{id: 0, value: ''}>}
     */
    data: [],
    selectedValue: '',
    setSearchBar: false,
    onSelect: (item) => {},
    enabled: true,

    /**
     * @type {React.Ref}
     * @description Used to prevent scroll page when scrolling list
     */
    pageRef: null,
    ignoreWarning: false
}

class ComboBox extends React.Component {
    state = {
        parent: {
            width: 0,
            height: 0,
            x: 0, y: 0
        },
        anim: new Animated.Value(0),

        data: [],
        selectionMode: false,
        selectedValue: '',
        search: ''
    }

    componentDidMount() {
        this.flatlistRef = null;
        this.refreshSearch();
    }

    componentDidUpdate(prevProps) {
        if (prevProps.data != this.props.data) {
            this.refreshSearch();
        }
        if (this.state.selectedValue != this.props.selectedValue) {
            this.setState({ selectedValue: this.props.selectedValue });
        }
    }

    onLayout = (event) => {
        const  { x: _x, y: _y, width: _width, height: _height } = this.state.parent;
        const  { x, y, width, height } = event.nativeEvent.layout;
        if (x !== _x || y !== _y || width !== _width || height !== _height) {
            const parent = { x: x, y: y, width: width, height: height };
            this.setState({ parent: parent });
        }
    }

    openSelection = () => {
        if (!this.props.enabled) return;

        // Disable parent scroll
        if (this.props.pageRef !== null) {
            this.props.pageRef.DisableScroll();
        } else if (!this.props.ignoreWarning) {
            console.warn('ComboBox: pageRef is null');
        }

        // Scroll to top
        if (this.flatlistRef !== null) {
            this.flatlistRef.scrollToOffset({ offset: 0, animated: false });
        }

        // Open selection
        SpringAnimation(this.state.anim, 1).start();
        this.setState({ selectionMode: true });
    }
    closeSelection = () => {
        if (this.props.pageRef !== null) {
            this.props.pageRef.EnableScroll();
        }
        SpringAnimation(this.state.anim, 0).start();
        this.setState({ selectionMode: false });
    }
    resetSelection = () => {
        if (!this.props.enabled) return;
        this.props.onSelect(null);
        this.setState({ selectedValue: '' });
    }

    refreshSearch = (text = '') => {
        if (text.length > 0) {
            const newDate = this.props.data.filter(item => item.value.toLowerCase().includes(text.toLowerCase()));
            this.setState({ data: newDate, search: text });
        } else {
            this.setState({ data: this.props.data, search: text });
        }
    }

    renderSearchBar = () => (
        <View style={styles.parentSearchBar}>
            <Input
                label={langManager.curr['modal']['search']}
                text={this.state.search}
                onChangeText={this.refreshSearch}
            />
        </View>
    )

    renderItem = ({ item }) => {
        const { id, value } = item;
        const onPress = () => {
            this.props.onSelect(item);
            this.setState({ selectedValue: value })
            this.closeSelection();
        };
        return (
            <TouchableHighlight
                style={styles.item}
                onPress={onPress}
                underlayColor={themeManager.GetColor('backgroundTransparent')}
            >
                <Text style={styles.itemText}>{value}</Text>
            </TouchableHighlight>
        )
    }
    renderChevron = () => {
        const angle = this.state.anim.interpolate({ inputRange: [0, 1], outputRange: ['0deg', '180deg'] });

        return this.props.enabled && (
            <View style={styles.chevron} pointerEvents='none'>
                <Animated.View style={{ transform: [{ rotateX: angle }] }}>
                    <Icon icon='chevron' size={20} angle={-90} />
                </Animated.View>
            </View>
        )
    }
    render() {
        const  { x, y, width, height } = this.state.parent;
        const animValue = this.state.anim.interpolate({ inputRange: [0, 1], outputRange: [y+height+20, y+height] });
        const overlayPos = [styles.overlay, {
            width: width,
            opacity: this.state.anim,
            transform: [{ translateX: x }, { translateY: animValue }],
            backgroundColor: themeManager.GetColor('backgroundGrey')
        }];
        const header = this.props.setSearchBar ? this.renderSearchBar : null;

        return (
            <>
                {/* Component (Input for selection, chevron and button for ripple + events) */}
                <View style={[styles.parent, this.props.style]} onLayout={this.onLayout}>
                    <Input label={this.props.title} text={this.state.selectedValue} active={this.state.selectionMode} pointerEvents='none' />
                    <this.renderChevron />
                    <Button style={styles.hoverButton} onPress={this.openSelection} onLongPress={this.resetSelection} rippleColor='white' />
                </View>

                {/* Overlay (black opacity + back event) */}
                {this.state.selectionMode && <View style={styles.overlayBackground} onTouchStart={this.closeSelection} />}

                {/* Content (flatlist with elements) */}
                <Animated.View style={overlayPos} pointerEvents={this.state.selectionMode ? 'auto': 'none'}>
                    <FlatList
                        ref={ref => this.flatlistRef = ref}
                        ListHeaderComponent={header}
                        data={this.state.data}
                        renderItem={this.renderItem}
                        keyExtractor={(item, index) => 'i-' + index}
                    />
                </Animated.View>
            </>
        );
    }
}

ComboBox.prototype.props = ComboBoxProps;
ComboBox.defaultProps = ComboBoxProps;

const styles = StyleSheet.create({
    overlay: {
        position: 'absolute',
        top: 0,
        left: 0,
        width: 128,
        maxHeight: 256,
        borderBottomLeftRadius: 8,
        borderBottomRightRadius: 8,
        zIndex: 1000,
        elevation: 1000
    },
    overlayBackground: {
        position: 'absolute',
        top: '-1000%',
        left: '-1000%',
        right: '-1000%',
        bottom: '-1000%',
        backgroundColor: '#00000055',
        zIndex: 900,
        elevation: 900
    },
    hoverButton: {
        position: 'absolute',
        width: 'auto',
        height: 'auto',
        top: 2,
        left: 2,
        right: 2,
        bottom: 2,
        borderRadius: 2,
        zIndex: 800,
        elevation: 800
    },

    parent: {
        width: '100%'
    },
    chevron: {
        position: 'absolute',
        top: 0,
        right: 12,
        bottom: 0,
        justifyContent: 'center'
    },

    item: {
        padding: 12
    },
    itemText: {
        textAlign: 'left'
    },
    parentSearchBar: {
        padding: 8
    }
});

export default ComboBox;