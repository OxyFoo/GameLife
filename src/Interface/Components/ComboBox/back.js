import * as React from 'react';
import { Animated } from 'react-native';

import { FormatForSearch } from 'Utils/String';
import { SpringAnimation } from 'Utils/Animations';
import { GetAbsolutePosition } from 'Utils/UI';

/**
 * @typedef {import('react-native').View} View
 * @typedef {import('react-native').ViewStyle} ViewStyle
 * @typedef {import('react-native').StyleProp<ViewStyle>} StyleProp
 * @typedef {import('react-native').LayoutRectangle} LayoutRectangle
 *
 * @typedef {import('react-native').FlatList} FlatList
 * @typedef {import('Managers/ThemeManager').ThemeColor} ThemeColor
 *
 * @typedef {object} ComboBoxItem
 * @property {number | string} key
 * @property {string} value
 *
 * @typedef {Object} ComboBoxPropsType
 * @property {StyleProp} style
 * @property {StyleProp} inputStyle
 * @property {number} maxContentHeight
 * @property {string} title
 * @property {ThemeColor} activeColor
 * @property {Array<ComboBoxItem>} data
 * @property {string} selectedValue
 * @property {boolean} enableSearchBar
 * @property {(item: ComboBoxItem | null) => void} onSelect
 * @property {boolean} enabled
 * @property {boolean} hideChevron
 */

/** @type {ComboBoxPropsType} */
const ComboBoxProps = {
    style: {},
    inputStyle: {},
    maxContentHeight: 256,
    title: 'Title',
    activeColor: 'main1',
    data: [],
    selectedValue: '',
    enableSearchBar: false,
    onSelect: () => {},
    enabled: true,
    hideChevron: false
};

class ComboBoxBack extends React.Component {
    state = {
        /** @type {LayoutRectangle} */
        parent: {
            x: 0,
            y: 0,
            width: 0,
            height: 0
        },

        anim: new Animated.Value(0),

        data: this.props.data,
        selectionMode: false,
        search: ''
    };

    /** @type {React.RefObject<View>} */
    refParent = React.createRef();

    /** @type {React.RefObject<FlatList>} */
    refFlatlist = React.createRef();

    /**
     * @param {ComboBoxProps} nextProps
     * @param {ComboBoxBack['state']} nextState
     */
    shouldComponentUpdate(nextProps, nextState) {
        return (
            this.props.data !== nextProps.data ||
            this.props.selectedValue !== nextProps.selectedValue ||
            this.props.enabled !== nextProps.enabled ||
            this.state.selectionMode !== nextState.selectionMode ||
            this.state.search !== nextState.search
        );
    }

    /** @param {ComboBoxProps} prevProps */
    componentDidUpdate(prevProps) {
        // Data changed, update data and reset search
        if (prevProps.data !== this.props.data) {
            this.setState({
                data: this.props.data,
                search: ''
            });
        }
    }

    onPress = () => {
        if (!this.props.enabled || this.refParent.current === null) {
            return;
        }

        if (this.state.selectionMode) {
            this.closeSelection();
            return;
        }

        // Scroll to top
        this.refFlatlist.current?.scrollToOffset({
            offset: 0,
            animated: false
        });

        // Open selection
        GetAbsolutePosition(this.refParent).then((rect) => {
            this.setState({ parent: rect, selectionMode: true }, () => {
                SpringAnimation(this.state.anim, 1).start();
            });
        });
    };

    closeSelection = () => {
        SpringAnimation(this.state.anim, 0).start();
        this.setState({ selectionMode: false });
    };

    resetSelection = () => {
        if (!this.props.enabled) {
            return;
        }
        this.props.onSelect(null);
        if (this.state.selectionMode) {
            this.closeSelection();
        }
    };

    refreshSearch = (text = '') => {
        const textLowerCase = FormatForSearch(text);
        this.setState({
            data: this.props.data.filter((item) => FormatForSearch(item.value).includes(textLowerCase)),
            search: text
        });
    };

    /** @param {ComboBoxItem} item */
    onItemPress = (item) => {
        this.props.onSelect(item);
        this.closeSelection();
    };
}

ComboBoxBack.prototype.props = ComboBoxProps;
ComboBoxBack.defaultProps = ComboBoxProps;

export default ComboBoxBack;
