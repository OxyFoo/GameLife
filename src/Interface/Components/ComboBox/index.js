import * as React from 'react';
import { View, Animated, TouchableHighlight, FlatList, Modal } from 'react-native';

import styles from './style';
import ComboBoxBack from './back';
import themeManager from 'Managers/ThemeManager';
import langManager from 'Managers/LangManager';

import { Text } from '../Text';
import { Button } from '../Button';
import { Icon } from '../Icon';
import { InputText } from '../InputText';

/**
 * @typedef {import('react-native').ViewStyle} ViewStyle
 * @typedef {import('react-native').StyleProp<ViewStyle>} StyleProp
 * @typedef {import('./back').ComboBoxItem} ComboBoxItem
 */

class ComboBox extends ComboBoxBack {
    render() {
        const { selectionMode } = this.state;

        return (
            <>
                {this.renderElement()}

                <Modal visible={selectionMode} transparent={true} animationType='fade'>
                    {this.renderOverlay()}
                    {this.renderContent()}
                </Modal>
            </>
        );
    }

    renderElement = () => {
        const { style, inputStyle, title, activeColor, enabled, selectedValue, hideChevron } = this.props;
        const { anim, selectionMode } = this.state;

        const angle = anim.interpolate({
            inputRange: [0, 1],
            outputRange: ['0deg', '180deg']
        });

        return (
            <View ref={this.refParent} style={[styles.parentContent, style]}>
                {/* Button for interaction (open combobox) */}
                <Button
                    testID='combobox-button'
                    style={styles.hoverButton}
                    appearance='uniform'
                    color='transparent'
                    onPress={this.onPress}
                    onLongPress={this.resetSelection}
                    onTouchMove={this.closeSelection}
                />

                {/* InputText to show result */}
                <InputText
                    style={inputStyle}
                    label={title}
                    value={selectedValue}
                    activeColor={activeColor}
                    forceActive={selectionMode}
                    pointerEvents='none'
                />

                {/* Chevron icon */}
                {enabled && !hideChevron && (
                    <View style={styles.chevron} pointerEvents='none'>
                        <Animated.View style={{ transform: [{ rotateX: angle }] }}>
                            <Icon icon='chevron' color={selectionMode ? activeColor : 'border'} size={20} angle={-90} />
                        </Animated.View>
                    </View>
                )}
            </View>
        );
    };

    renderOverlay = () => {
        const { selectionMode } = this.state;

        if (!selectionMode) {
            return null;
        }

        return <View style={styles.overlayBackground} onTouchStart={this.closeSelection} />;
    };

    renderContent = () => {
        const { enableSearchBar: setSearchBar, maxContentHeight: maxHeight, activeColor } = this.props;
        const { parent, anim, data, selectionMode } = this.state;
        const { x, y, width, height } = parent;

        const animValue = anim.interpolate({
            inputRange: [0, 1],
            outputRange: [-60, 0]
        });

        /** @type {StyleProp} */
        const overlayStyle = {
            width: width,
            maxHeight: maxHeight,
            transform: [{ translateX: x }, { translateY: y + height }]
        };

        /** @type {StyleProp} */
        const borderFixStyle = {
            opacity: anim,
            transform: [{ translateX: x }, { translateY: y + height }],
            borderColor: themeManager.GetColor(activeColor)
        };

        /** @type {StyleProp} */
        const panelStyle = {
            opacity: anim,
            transform: [{ translateY: animValue }],
            borderColor: themeManager.GetColor(activeColor)
        };

        return (
            <>
                <Animated.View style={[styles.borderFix, borderFixStyle, { width }]} pointerEvents={'none'} />

                <Animated.View
                    style={[styles.overlayParent, overlayStyle]}
                    pointerEvents={selectionMode ? 'auto' : 'none'}
                >
                    <Animated.View style={[styles.overlayPanel, panelStyle]}>
                        <FlatList
                            ref={this.refFlatlist}
                            ListHeaderComponent={
                                !setSearchBar ? null : (
                                    <View style={styles.parentSearchBar}>
                                        <InputText
                                            style={styles.search}
                                            containerStyle={styles.searchContainer}
                                            label={langManager.curr['modal']['search']}
                                            value={this.state.search}
                                            onChangeText={this.refreshSearch}
                                        />
                                    </View>
                                )
                            }
                            style={[
                                styles.overlayContent,
                                {
                                    backgroundColor: themeManager.GetColor('backgroundGrey')
                                }
                            ]}
                            data={data}
                            renderItem={this.renderItem}
                            keyExtractor={(item, index) => `i-${item.key}-${index}`}
                        />
                    </Animated.View>
                </Animated.View>
            </>
        );
    };

    /** @param {{ item: ComboBoxItem }} param0 */
    renderItem = ({ item }) => {
        const backgroundColor = themeManager.GetColor('backgroundTransparent');

        return (
            <TouchableHighlight
                style={styles.item}
                onPress={() => this.onItemPress(item)}
                underlayColor={backgroundColor}
            >
                <Text style={styles.itemText}>{item.value}</Text>
            </TouchableHighlight>
        );
    };
}

export { ComboBox };
