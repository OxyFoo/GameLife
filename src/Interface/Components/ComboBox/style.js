import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    parentContent: {
        width: '100%',
        zIndex: 1010,
        elevation: 1010
    },
    input: {
        borderRadius: 8
    },

    overlayParent: {
        position: 'absolute',
        top: 0,
        left: 0,
        width: 128,

        overflow: 'hidden',
        zIndex: 1000,
        elevation: 1000
    },
    overlayPanel: {
        marginBottom: 12,
        borderColor: '#ffffff',
        borderWidth: 1,
        borderTopWidth: 0,
        borderBottomLeftRadius: 8,
        borderBottomRightRadius: 8,
        overflow: 'hidden'
    },
    overlayContent: {
        borderBottomLeftRadius: 8,
        borderBottomRightRadius: 8
    },
    borderFix: {
        position: 'absolute',
        top: -12,
        left: 0,
        right: 0,
        height: 14,
        borderLeftWidth: 1,
        borderRightWidth: 1,
        zIndex: 1020,
        elevation: 1020
    },
    overlayBackground: {
        position: 'absolute',
        top: '-1000%',
        left: '-1000%',
        right: '-1000%',
        bottom: '-1000%',
        backgroundColor: '#00000055',
        zIndex: 900,
        elevation: 900
    },
    hoverButton: {
        position: 'absolute',
        width: 'auto',
        height: 'auto',
        top: 2,
        left: 2,
        right: 2,
        bottom: 2,
        borderRadius: 6,
        zIndex: 800,
        elevation: 800
    },

    chevron: {
        position: 'absolute',
        top: 0,
        right: 12,
        bottom: 0,
        justifyContent: 'center'
    },

    item: {
        padding: 12
    },
    itemText: {
        textAlign: 'left'
    },
    parentSearchBar: {
        padding: 8
    },
    search: {
        height: 36
    }
});

export default styles;
