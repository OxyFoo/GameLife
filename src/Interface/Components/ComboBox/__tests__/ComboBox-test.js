import 'react-native';
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

// Graphical components
import ComboBox from '../index';

const options = [
    { key: 'Option 1', value: 'Option 1' },
    { key: 'Option 2', value: 'Option 2' },
    { key: 'Option 3', value: 'Option 3' }
];

describe('[Component] Input', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<ComboBox />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render the options correctly', () => {
        const { getByText, queryByText } = render(
            <ComboBox data={options} />
        );

        options.forEach((option) => {
            const optionElement = getByText(option.value);
            expect(optionElement).toBeDefined();
        });

        const nonExistentOption = queryByText('Non-Existent Option');
        expect(nonExistentOption).toBeNull();
    });

    it('should call a callback function when an option is selected', () => {
        const onSelectMock = jest.fn();
        const { getByText, getByTestId } = render(
            <ComboBox data={options} onSelect={onSelectMock} />
        );

        const buttonCombobox = getByTestId('combobox-button');
        const optionElement = getByText('Option 2');

        // Open the dropdown
        fireEvent.press(buttonCombobox);

        // Select the option
        fireEvent.press(optionElement);

        expect(onSelectMock).toHaveBeenCalledWith(options[1]);
    });
});
