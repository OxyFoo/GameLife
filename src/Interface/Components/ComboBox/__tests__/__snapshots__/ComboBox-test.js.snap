// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Input should render correctly 1`] = `
<View
  style={
    [
      {
        "elevation": 1010,
        "width": "100%",
        "zIndex": 1010,
      },
      {},
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "disabled": false,
      }
    }
    accessible={true}
    borderColor="automatic"
    collapsable={false}
    fontColor="automatic"
    onLayout={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    pointerEvents="auto"
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderRadius": 8,
          "overflow": "hidden",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
          "width": "100%",
        },
        {
          "borderRadius": 6,
          "bottom": 2,
          "elevation": 800,
          "height": "auto",
          "left": 2,
          "position": "absolute",
          "right": 2,
          "top": 2,
          "width": "auto",
          "zIndex": 800,
        },
        null,
      ]
    }
    testID="combobox-button"
    throttleTime={250}
  >
    <View
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {},
          {
            "backgroundColor": "#00000000",
          },
        ]
      }
    />
    <View
      pointerEvents="none"
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "center",
          },
          {
            "justifyContent": "center",
            "opacity": 1,
          },
          {},
        ]
      }
    />
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        {
          "aspectRatio": 1,
          "backgroundColor": "#FFFFFF",
          "borderRadius": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transform": [
            {
              "translateX": 0,
            },
            {
              "translateY": 0,
            },
            {
              "scale": 0,
            },
          ],
          "width": 3,
        }
      }
    />
  </View>
  <View
    collapsable={false}
    onLayout={[Function]}
    pointerEvents="none"
    style={
      [
        {
          "borderRadius": 8,
          "width": "100%",
        },
        {
          "backgroundColor": "#00000000",
          "borderColor": "#D0D0D0",
          "borderWidth": 1.2,
          "opacity": 1,
          "paddingRight": 0,
        },
        {},
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#0E1247",
          "height": 3,
          "left": 6,
          "position": "absolute",
          "top": -2,
          "transform": [
            {
              "scaleX": 0,
            },
          ],
          "width": 0,
        }
      }
    />
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "left": 0,
          "overflow": "scroll",
          "paddingHorizontal": 4,
          "position": "absolute",
          "top": 0,
          "transform": [
            {
              "translateX": -0,
            },
            {
              "translateY": -0,
            },
            {
              "scale": 1,
            },
            {
              "translateX": 0,
            },
            {
              "translateY": 0,
            },
            {
              "translateX": 16,
            },
            {
              "translateY": -2,
            },
          ],
        }
      }
    >
      <Text
        onLayout={[Function]}
        style={
          [
            {
              "fontFamily": "Hind Vadodara",
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
            },
            {
              "color": "#ECECEC",
              "fontSize": 16,
            },
            {},
          ]
        }
      >
        Title
      </Text>
    </View>
    <TextInput
      autoComplete="off"
      autoCorrect={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#B0B0B0"
      selectionColor="white"
      style={
        [
          {
            "fontSize": 18,
            "maxHeight": 256,
            "minHeight": 12,
            "paddingHorizontal": 24,
            "paddingVertical": 16,
          },
          {
            "verticalAlign": "middle",
          },
          {
            "color": "#ECECEC",
          },
          {},
        ]
      }
      testID="textInput"
      textContentType="none"
      value=""
    />
  </View>
  <View
    pointerEvents="none"
    style={
      {
        "bottom": 0,
        "justifyContent": "center",
        "position": "absolute",
        "right": 12,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "rotateX": "0deg",
            },
          ],
        }
      }
    >
      <View
        style={
          [
            {
              "height": 20,
              "width": 20,
            },
            {},
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={20}
          bbWidth={20}
          color="#999EB5"
          fill="#999EB5"
          focusable={false}
          height={20}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 20,
                "width": 20,
              },
            ]
          }
          transform={
            [
              {
                "rotate": -1.5707963267948966,
              },
            ]
          }
          vbHeight={185.343}
          vbWidth={185.343}
          width={20}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4288257717,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M51.707 185.343a10.692 10.692 0 0 1-7.593-3.149 10.724 10.724 0 0 1 0-15.175l74.352-74.347L44.114 18.32c-4.194-4.194-4.194-10.987 0-15.175 4.194-4.194 10.987-4.194 15.18 0l81.934 81.934c4.194 4.194 4.194 10.987 0 15.175l-81.934 81.939a10.678 10.678 0 0 1-7.587 3.15z"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;
